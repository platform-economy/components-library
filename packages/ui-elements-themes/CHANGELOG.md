# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
