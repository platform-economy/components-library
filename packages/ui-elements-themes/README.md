# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
