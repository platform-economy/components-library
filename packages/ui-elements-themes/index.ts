export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
