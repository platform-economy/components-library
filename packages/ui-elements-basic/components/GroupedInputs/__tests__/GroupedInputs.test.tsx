import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
/*
A set of utilities for testing Styled Components with Jest.
This package improves the snapshot testing experience and
provides a brand new matcher to make expectations on the style rules.
*/
import 'jest-styled-components';
import { withTestTheme } from '@relayr/ui-elements-themes';

// Component
import { GroupedInputs, countRatio } from '../index';

describe('ui-elements-basic', () => {
  describe('Components', () => {
    describe('GroupedInputs', () => {

      it('should render with default state', () => {
        const tree = TestRenderer
          .create(withTestTheme(
          <GroupedInputs
            componentsLeft={<input type="text"/>}
            componentsRight={<input type="text"/>}
            labelLeft="test"
            labelRight="test"
          />))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should render with provided ratio', () => {
        const tree = TestRenderer
          .create(withTestTheme(
          <GroupedInputs
            dividerRatio={30}
            componentsLeft={<input type="text"/>}
            componentsRight={<input type="text"/>}
            labelLeft="test"
            labelRight="test"
          />))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should properly return ratio percentage', () => {
        expect(countRatio(10)).toBe('10%');
        expect(countRatio()).toBe('50%');
      });
    });
  });
});
