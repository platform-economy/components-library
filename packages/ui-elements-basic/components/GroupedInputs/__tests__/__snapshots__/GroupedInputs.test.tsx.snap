// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-basic Components GroupedInputs should render with default state 1`] = `
.c1 {
  min-height: 28px;
  border-left: solid 1px #divider;
  margin: auto 1px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: auto;
  height: auto;
  color: #text;
  border: none;
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0.c0 label {
  word-break: normal;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

.c0 label:first-child {
  width: 50%;
}

.c0 label:last-child {
  width: 50%;
}

.c0 input {
  box-shadow: none;
  background-color: transparent;
}

.c0 div.groupBox {
  width: 100%;
  height: auto;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: 0 1px 3px 0 #shadow;
  background-color: #foreground;
}

.c0 .groupBox>input:first-child {
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
}

.c0 .groupBox>input:last-child {
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
}

.c0 .groupBox>div:last-child {
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
}

<div
  className="c0"
>
  test
  test
  <div
    className="groupBox"
  >
    <input
      type="text"
    />
    <div
      className="c1"
    />
    <input
      type="text"
    />
  </div>
</div>
`;

exports[`ui-elements-basic Components GroupedInputs should render with provided ratio 1`] = `
.c1 {
  min-height: 28px;
  border-left: solid 1px #divider;
  margin: auto 1px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: auto;
  height: auto;
  color: #text;
  border: none;
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0.c0 label {
  word-break: normal;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

.c0 label:first-child {
  width: 30%;
}

.c0 label:last-child {
  width: 70%;
}

.c0 input {
  box-shadow: none;
  background-color: transparent;
}

.c0 div.groupBox {
  width: 100%;
  height: auto;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: 0 1px 3px 0 #shadow;
  background-color: #foreground;
}

.c0 .groupBox>input:first-child {
  -webkit-flex-basis: 30%;
  -ms-flex-preferred-size: 30%;
  flex-basis: 30%;
}

.c0 .groupBox>input:last-child {
  -webkit-flex-basis: 70%;
  -ms-flex-preferred-size: 70%;
  flex-basis: 70%;
}

.c0 .groupBox>div:last-child {
  -webkit-flex-basis: 70%;
  -ms-flex-preferred-size: 70%;
  flex-basis: 70%;
}

<div
  className="c0"
>
  test
  test
  <div
    className="groupBox"
  >
    <input
      type="text"
    />
    <div
      className="c1"
    />
    <input
      type="text"
    />
  </div>
</div>
`;
